### PR TITLE
React-i18next support for mainapp and plugins + typescript support for typeing + HMR

### PR DESCRIPTION
**Pull request by C3ntraX**

### CODE REVIEW

File: src/echo-app/package.json [hunk 1]
```diff
@@ -27,6 +27,7 @@
     "@electron/remote": "^2.0.12",
     "@libsql/client": "^0.4.0-pre.2",
     "body-parser": "^1.20.2",
+    "character-plugin": "file:../echo-plugin-examples/character-plugin",
```
Comment (zach-herridge) [inline src/echo-app/package.json:30]: Need to remove these local plugin or the prod build will break.

**Comment by zach-herridge** [comment]
We probably don't need to be adding translations to the the example plugins, they make it more complicated and for most people the first thing they will do after copying an example plugin is remove the i18next

**Comment by zach-herridge** [comment]
Creating an example that is just for how to setup translations would be good though

**Comment by C3ntraX** [comment]
You want to create a new plugin only as example for translations?

**Comment by C3ntraX** [comment]
I think it would be the best to only remove the useTranslation hooks and empty the used key. Additional, we could write a guide, how to use this buildin feature?

**Comment by zach-herridge** [comment]
Yeah that's fine, it won't break if someone does remove them right?

**Comment by C3ntraX** [comment]
If someone removes it, its removed. Does not matter.
Remove `locales` ,`types` and `config` removes all

**Comment by C3ntraX** [comment]
No script dependents or any other

**Review by zach-herridge — APPROVED** [review]
Cool sounds good

**Pull request merged**